src/core/trtserver.cc

```diff
@@ -852,17 +851,16 @@ TRTSERVER_ServerInferAsync(
     infer_response_provider = del_response_provider;
   }
 
-  auto request_status = std::make_shared<ni::RequestStatus>();
-  lserver->HandleInfer(
-      request_status.get(), lprovider->Backend(), infer_request_provider,
-      infer_response_provider, infer_stats,
-      [infer_stats, timer, request_status, infer_response_provider, server,
-       complete_fn, userp]() mutable {
+  lserver->Infer(
```

Comment:
check return value, trigger `complete_fn` is not okay since the lambda function will not be called.